### PR TITLE
feat: reorganize events based on new rules

### PR DESCRIPTION
**Pull request by adjkant**
Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) and start with a valid [type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type).

## Background

This PR updates the naming and organization of enums to match the new rule set

## Changes

- Add `Interface` prefixes to specific events
- Separate general swapping/NFT events
- Break out interface events for testing

## Testing
- Implemented in https://github.com/Uniswap/interface/pull/5505 and https://github.com/Uniswap/docs/pull/519

#### Before merging, please ensure the following:

- [x] All events have been tested in their consuming package
- [x] This PR is titled accordingly to the scope of the events change, if relevant (`fix` for new event info in an existing section, `feat` for a new product/section)

BREAKING CHANGE: Reorganizes events based on product and feature and removes CommonJS support

### CODE REVIEW

File: src/swap/index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  SWAP_MODIFIED_IN_WALLET = 'Swap Modified in Wallet',
+  SWAP_PRICE_UPDATE_ACKNOWLEDGED = 'Swap Price Update Acknowledged',
+  SWAP_QUOTE_RECEIVED = 'Swap Quote Received',
+  SWAP_SIGNED = 'Swap Signed',
+  SWAP_SUBMITTED_BUTTON_CLICKED = 'Swap Submit Button Clicked',
+  SWAP_TOKENS_REVERSED = 'Swap Tokens Reversed',
+  SWAP_TRANSACTION_COMPLETED = 'Swap Transaction Completed',
+}
+
+export enum SwapPriceUpdateUserResponse {
```
Comment (dannydaniil) [inline src/swap/index.ts:17]: Wondering why we did not split these into different files either

**Review by dannydaniil — APPROVED** [review]
LGTM

File: src/nft/index.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  NFT_SELL_SELECTED = 'NFT Sell Selected',
+  NFT_SELL_START_LISTING = 'NFT Sell Start Listing',
+  NFT_TRENDING_ROW_SELECTED = 'Trending Row Selected',
+}
+
+/**
+ * Known Filter Types for NFTs
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export enum NFTFilterTypes {
```
[the remark this answers is not in the document]
Comment (vm) [reply src/nft/index.ts:27]: i think you should keep it consistent for nft.

File: src/primitives.ts [hunk 3]
```diff
@@ -1,59 +1,16 @@
 /**
- * Event names that can occur in the application.
- *
- * Subject to change as new features are added and new events are defined
- * and logged.
+ * General event names that can be reused across products.
  */
-export enum EventName {
-  APPROVE_TOKEN_TXN_SUBMITTED = 'Approve Token Transaction Submitted',
+export enum BasicEventName {
```
Comment (vm) [inline src/primitives.ts:4]: Basic is a bit unclear. Any other ideas on names?

File: src/swap/index.ts (hunk 1 above)
Comment (vm) [reply src/swap/index.ts:17]: agreed i prefer diff files here

**Review by vm — COMMENTED** [review]
Please wait to merge this change until we finish the launches next week. I can help rebase if necessary but I don't want to have a major version locked in while we need to add more events coming up.

File: src/primitives.ts (hunk 3 above)
Comment (adjkant) [reply src/primitives.ts:4]: Thoughts on `PrimitiveEventName`?

Comment (vm) [reply src/primitives.ts:4]: `SharedEventName`?

**Comment by adjkant** [comment]
> Please wait to merge this change until we finish the launches next week. I can help rebase if necessary but I don't want to have a major version locked in while we need to add more events coming up.

That works for me, def not a rush here especially with launches happening

**Review by lynnshaoyu — APPROVED** [review]

**Pull request merged**